consensus/src/chained_bft/network.rs

```diff
@@ -55,17 +55,20 @@ impl<T: Payload> BlockRetrievalResponse<T> {
                 self.blocks.len(),
             ));
         }
-        for block in self.blocks.iter() {
```

Comment:
If the main goal is to remove the mut, it might be simpler just to copy the block_id rather than the try_fold and map

Reply:
See also the error message interversion.
I can copy the block if people find this simpler.